### DISCUSSION TITLE
Genes don't show up when plotting

### DISCUSSION THREAD

**Discussion by zrcjessica** [stuart-lab/signac#746] [Q&A] [answered]
Hello,

I'm trying to plot some results and I've noticed that my Genes track is always empty. However, when I run `Annotation(dat)` I do see an annotation for my Signac object:

```
                     seqnames        ranges strand |            gene_id
                        <Rle>     <IRanges>  <Rle> |        <character>
  ENSRNOG00000002314       10   28243-72912      + | ENSRNOG00000002314
  ENSRNOG00000022305       10 549883-672196      - | ENSRNOG00000022305
  ENSRNOG00000053230       10 730247-751061      + | ENSRNOG00000053230
  ENSRNOG00000057880       10 764421-859184      + | ENSRNOG00000057880
  ENSRNOG00000058007       10 860521-896938      - | ENSRNOG00000058007
  ENSRNOG00000056248       10 908806-953474      + | ENSRNOG00000056248
                        gene_name   gene_biotype seq_coord_system
                      <character>    <character>      <character>
  ENSRNOG00000002314 LOC103689943 protein_coding       chromosome
  ENSRNOG00000022305        Abcc1 protein_coding       chromosome
  ENSRNOG00000053230        Fopnl protein_coding       chromosome
  ENSRNOG00000057880        Myh11 protein_coding       chromosome
  ENSRNOG00000058007         Nde1 protein_coding       chromosome
  ENSRNOG00000056248        Marf1 protein_coding       chromosome
                                description      gene_id_version       symbol
                                <character>          <character>  <character>
  ENSRNOG00000002314 meiosis arrest femal.. ENSRNOG00000002314.7 LOC103689943
  ENSRNOG00000022305 ATP binding cassette.. ENSRNOG00000022305.5        Abcc1
  ENSRNOG00000053230 FGFR1OP N-terminal l.. ENSRNOG00000053230.1        Fopnl
  ENSRNOG00000057880 myosin heavy chain 1.. ENSRNOG00000057880.1        Myh11
  ENSRNOG00000058007 nudE neurodevelopmen.. ENSRNOG00000058007.1         Nde1
  ENSRNOG00000056248 meiosis regulator an.. ENSRNOG00000056248.1        Marf1
                      entrezid
                        <list>
  ENSRNOG00000002314 103689943
  ENSRNOG00000022305     24565
  ENSRNOG00000053230    360461
  ENSRNOG00000057880     24582
  ENSRNOG00000058007     83836
  ENSRNOG00000056248    170946
  -------
  seqinfo: 23 sequences from Rnor_6.0 genome
```

I used GRanges to make sure that the regions I was trying to plot overlapped genes in the Annotation, but my Genes track in my coverage plots are still empty. I do notice I get various warning messages when I try to plot, e.g.: 
`Scale for 'x' is already present. Adding another scale for 'x', which will replace the existing scale.`

`Warning message: "Removed 1 rows containing missing values (position_stack)."`

Any idea what's wrong? Help much appreciated!

**Comment by timoast** [comment, marked as answer]
I think the issue here is that the gene annotations in your object do not have a "type" column, which is needed by Signac. We will work on updating the documentation here so it's clear what information is needed in the gene annotations.

See also https://github.com/timoast/signac/issues/656#issuecomment-867813864

**Reply by zrcjessica** [reply]
Hi @timoast,

I made sure that my annotation had a type column and it resolved the issue. However, I still notice that often times when I plot ranges that should include a gene or part of a gene body, the gene still won't show up in the plot. Alternatively, a gene will show up but not the name of the gene. 

I also noticed that I sometimes get this error when I try to plot a region: 
```
Error: Aesthetics must be either length 1 or the same as the data (253): label
```
In the example where I got this error, I expanded one of the regions in the Links by 10kb on each side. 

Additionally, I have also tried giving the function a gene instead of coordinates, e.g.: `CoveragePlot(dat, region = "Tmem212")` but I get this error:
```
Error: logical subscript contains NAs
```
I have checked to make sure that there is a gene type record for this gene in the annotation for my Signac object. 

Do you know what might be causing any of these errors? Thanks!

**Reply by timoast** [reply]
Hi @zrcjessica, could you open an issue for this, and if possible share the gene annotation object that gives an error? Thanks!

**Reply by zrcjessica** [reply]
Done! #771
